### PR TITLE
[#5739] feat(model-catalog): Implement the model catalog logic

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This PR adds the model catalog implementation.

### Why are the changes needed?

This is a part of work to support model management in Gravitino.

Fix: #5739 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Added UTs to cover the changes.

### CODE REVIEW

**Comment by jerryshao** [comment]
@yuqi1129 @mchades @xloya can you please help to review this PR? Thanks.

File: catalogs/catalog-model/build.gradle.kts [hunk 1]
```diff
@@ -40,11 +40,17 @@ dependencies {
     exclude(group = "*")
   }
 
-  compileOnly(libs.guava)
-
+  implementation(libs.guava)
   implementation(libs.slf4j.api)
 
+  testImplementation(project(":clients:client-java"))
+  testImplementation(project(":integration-test-common", "testArtifacts"))
```
Comment (mchades) [inline catalogs/catalog-model/build.gradle.kts:47]: Why is this necessary?

**Review by mchades — COMMENTED** [review]
One confusion not about this PR:
I noticed that we have the `listModelVersions` operation, but why don't we have a `listModelAlias` method?

File: catalogs/catalog-model/build.gradle.kts (hunk 1 above)
Comment (jerryshao) [reply catalogs/catalog-model/build.gradle.kts:47]: It is not used currently, will be used when integration test is added.

**Comment by jerryshao** [comment]
> I noticed that we have the listModelVersions operation, but why don't we have a listModelAlias method?

There's no specific reason why this "listModelAliases" API is not added, currently there's no strong demand on it. I think we can add it later on when we require this API.

**Review by mchades — APPROVED** [review]
LGTM.
